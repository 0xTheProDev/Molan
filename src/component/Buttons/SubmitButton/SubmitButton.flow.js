// @flow
export type SubmitButtonProps = {
  SubmitButtonText: string,
  onSubmitButtonClick: (event: MouseEvent) => void,
};
