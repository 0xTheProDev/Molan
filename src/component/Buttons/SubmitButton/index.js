// @flow
export { SubmitButton, SubmitButton as default } from './SubmitButton';
export { SubmitButtonProps } from './SubmitButton.flow';
