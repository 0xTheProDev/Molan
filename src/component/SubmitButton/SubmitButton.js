// @flow
import React, { PureComponent } from 'react';
import { Button } from 'semantic-ui-react';
import { NotificationManager } from 'react-notifications';

type SubmissionData = {
  id: number,
  language: string,
  source: string,
  haveInput: string,
  input?: string,
};

type SubmitButtonProps = {
  lang: string,
  code: string,
  checked: boolean,
  input?: string,
  service: {
    status?: boolean,
  },
  submit: (submissionData: SubmissionData) => void,
  onCallback: () => void,
};

export class SubmitButton extends PureComponent<SubmitButtonProps> {
  onClick = () => {
    const {
      lang,
      code,
      checked,
      input,
      service,
      submit,
      onCallback,
    } = this.props;

    if (navigator.onLine && service.hasOwnProperty('status') && service.status === true) {
      const formData: SubmissionData = {
        id: Date.now(),
        language: lang,
        source: code,
        haveInput: checked,
        input: input,
      };
      submit(formData);
      onCallback();
    } else {
      NotificationManager.error('Please check back after several minutes', 'Service Unreachable');
    }
  };

  render() {
    return (
      <Button
        type='button'
        size='large'
        floated='right'
        color='green'
        onClick={this.onClick}
      >Submit</Button>
    );
  }
}
