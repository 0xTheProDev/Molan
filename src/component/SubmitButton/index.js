import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';

export default class SubmitButton extends Component {
    static propTypes = {
        onCallback: PropTypes.func.isRequired
    };

    onClick = () => {
        this.props.onCallback();
    };

    render() {
        return (
            <Button
              type='button'
              size='large'
              floated='right'
              color='green'
              onClick={this.onClick}
            >Submit</Button>
        );
    }
}