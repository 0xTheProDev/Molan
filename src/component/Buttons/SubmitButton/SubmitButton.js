// @flow
import React from 'react';
import { Button } from 'semantic-ui-react';
import { noop } from 'util/common';

import { SubmitButtonProps } from './SubmitButton.flow';

export function SubmitButton({ SubmitButtonText, onSubmitButtonClick }: SubmitButtonProps) {
  return (
    <Button size='large' floated='right' color='green' onClick={onSubmitButtonClick}>
      {SubmitButtonText}
    </Button>
  );
}

SubmitButton.defaultProps = {
  SubmitButtonText: 'Submit',
  onSubmitButtonClick: noop,
};
