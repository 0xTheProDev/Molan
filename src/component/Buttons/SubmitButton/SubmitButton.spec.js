import React from 'react';
import { Button } from 'semantic-ui-react';

import { SubmitButton } from './SubmitButton';

describe('Component::Buttons:SubmitButton::SubmitButton', () => {
  let
    onSubmitButtonClickStub,
    minProps;

  beforeAll(() => {
    onSubmitButtonClickStub = jest.fn();

    minProps = {
      SubmitButtonText: 'Submit',
      onSubmitButtonClick: onSubmitButtonClickStub,
    };
  });

  it('should render with base structure', () => {
    const wrapper = shallow(<SubmitButton {...minProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke callback when Button clicked', () => {
    const wrapper = shallow(<SubmitButton {...minProps} />);

    wrapper.find(Button).simulate('click');

    expect(onSubmitButtonClickStub).toHaveBeenCalled();
  });
});
