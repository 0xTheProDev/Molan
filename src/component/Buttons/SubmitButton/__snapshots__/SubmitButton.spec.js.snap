// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::Buttons:SubmitButton::SubmitButton should render with base structure 1`] = `
<Button
  as="button"
  color="green"
  floated="right"
  onClick={[MockFunction]}
  size="large"
>
  Submit
</Button>
`;
